### PR TITLE
Stl decomposer after fe

### PR DESCRIPTION
**Pull request by chukarsten**
### Pull Request Description
(replace this text with your description)

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py [hunk 1]
```diff
@@ -111,6 +131,8 @@ def _project_trend_and_seasonality(self, y):
             self.seasonal_period,
             self.frequency,
         )
+        pd.testing.assert_index_equal(y.index, projected_seasonality.index)
+        pd.testing.assert_index_equal(y.index, projected_trend.index)
```
Comment (eccabay) [inline evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py:135]: Not sure we want tests in our source code?

**Review by eccabay — APPROVED** [review]
Looks good, left small comments but certainly nothing blocking

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py [hunk 2]
```diff
@@ -220,6 +221,19 @@ def set_seasonal_period(self, X: pd.DataFrame, y: pd.Series):
         self.seasonal_period = self.determine_periodicity(X, y)
         self.parameters["seasonal_period"] = self.seasonal_period
 
+    def _choose_proper_index(self, y):
+        """Function that provides support for targets with integer and datetime indices."""
+        # TODO: Need to update this after we upgrade to Pandas 1.5+ and Int64Index is deprecated.
+        if isinstance(y.index, (Int64Index, pd.RangeIndex)):
```
Comment (jeremyliweishih) [inline evalml/pipelines/components/transformers/preprocessing/decomposer.py:227]: can you help me understand why we change the index depending on what the passed in index is?

File: evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py [hunk 3]
```diff
@@ -117,6 +117,7 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None) -> PolynomialDecomposer:
             period=self.seasonal_period,
         ).seasonal
         self.seasonality = self.seasonal[0 : self.seasonal_period]
+        self.trend = y - (y_detrended_with_time_index - self.seasonal) - self.seasonal
```
Comment (jeremyliweishih) [inline evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py:120]: is `(y_detrended_with_time_index - self.seasonal)` the constant?

File: evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py [hunk 4]
```diff
@@ -217,8 +239,9 @@ def transform(
                     index=truncated_y.index,
                 ),
             )
-
-        return X, y_in_sample.append(y_out_of_sample)
+        y_t = y_in_sample.append(y_out_of_sample)
+        y_t.index = original_index
```
Comment (jeremyliweishih) [inline evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py:244]: minor point: should we use `self.original_index` here for clarity? or set `self.original_index lower`?

**Review by jeremyliweishih — APPROVED** [review]
LGTM just some clarifying questions

File: evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeremyliweishih) [reply evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py:120]: I think this is correct 😄

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py [hunk 5]
```diff
@@ -220,6 +221,19 @@ def set_seasonal_period(self, X: pd.DataFrame, y: pd.Series):
...
+            index = self.original_index
```
Comment (ParthivNaresh) [inline evalml/pipelines/components/transformers/preprocessing/decomposer.py:228]: Does the PolynomialDecomposer have `self.original_index` to return?

File: evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py [hunk 6]
```diff
@@ -236,30 +264,44 @@ def inverse_transform(self, y_t: pd.Series) -> tuple[pd.DataFrame, pd.Series]:
...
-                + self.seasonal[left_index:right_index]
+            right_index = (
+                y_t.index[-1] + 1
+                if isinstance(y_t.index, (Int64Index, pd.RangeIndex))
+                else y_t.index[-1] + 1 * y_t.index.freq
+            )
+            trend = (
+                self.trend.reset_index(drop=True)[left_index:right_index]
+                if isinstance(y_t.index, (Int64Index, pd.RangeIndex))
+                else self.trend[left_index:right_index]
```
[the remark this answers is not in the document]
Comment (chukarsten) [reply evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py:286]: This was actually ok!

File: evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py (hunk 3 above)
Comment (chukarsten) [reply evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py:120]: That value should be the residual!

File: evalml/pipelines/components/transformers/preprocessing/decomposer.py (hunk 5 above)
Comment (chukarsten) [reply evalml/pipelines/components/transformers/preprocessing/decomposer.py:228]: Will fix this - have a separate branch/PR that is bringing PD up to complete parity.

**Pull request merged**